src/pymor/reductors/aaa.py

```diff
@@ -0,0 +1,411 @@
+                 L_rk_tol=1e-8):
+        if isinstance(samples_or_fom, TransferFunction) or hasattr(samples_or_fom, 'transfer_function'):
+            fom = samples_or_fom
+            if not isinstance(samples_or_fom, TransferFunction):
+                fom = fom.transfer_function
+            self.num_vars = 1 + fom.parameters.dim
+
+            assert len(sampling_values) == self.num_vars
+            self.parameters = fom.parameters
+            self.samples = np.empty([len(sv) for sv in sampling_values] + [fom.dim_output, fom.dim_input],
+                                    dtype=sampling_values[0].dtype)
+            for idx, vals in zip(np.ndindex(self.samples.shape[:-2]),
+                                 itertools.product(*sampling_values)):
+                params = {p: v for p, v in zip(fom.parameters, vals[1:])}
+                self.samples[idx] = fom.eval_tf(vals[0], mu=params)
+            if fom.dim_input == fom.dim_output == 1:
+                self.samples = self.samples.reshape(self.samples.shape[:-2])
+        else:
+            self.samples = samples_or_fom
+            self.num_vars = len(sampling_values)
+            self.parameters = {'p': self.num_vars-1}
+
+        self.__auto_init(locals())
+
+    def reduce(self, tol=1e-7, max_itpl=None):
+        """Reduce using p-AAA.
+
+        Parameters
+        ----------
+        tol
+            Convergence tolerance for relative error of `rom` over the set of samples.
+        max_itpl
+            Maximum number of interpolation points to use with respect to each
+            variable. Should be `None` or a list such that `self.num_vars == len(max_itpl)`.
+            If `None` `max_itpl[i]` will be set to `len(self.sampling_values[i]) - 1`.
+
+        Returns
+        -------
+        rom
+            Reduced |TransferFunction| model.
+        """
+        svs = self.sampling_values
+        samples = self.samples
+
+        # add complex conjugate samples
+        if self.conjugate:
+            s_conj_list = []
+            samples_conj_list = []
+            for i, s in enumerate(svs[0]):
+                if s.conj() not in svs[0]:
+                    s_conj_list.append(s.conj())
+                    samples_conj_list.append(samples[i, None].conj())
+            if s_conj_list:
+                svs[0] = np.append(svs[0], s_conj_list)
+                samples = np.vstack([samples] + samples_conj_list)
+
+        num_vars = len(svs)
+        max_samples = np.max(np.abs(samples))
+        rel_tol = tol * max_samples
+
+        # Transform samples for MIMO case
+        if len(samples.shape) != len(svs):
+            assert len(samples.shape) == len(svs) + 2
+            dim_input = samples.shape[-1]
+            dim_output = samples.shape[-2]
+            samples_T = np.empty(samples.shape[:-2], dtype=samples.dtype)
+            w = new_rng(0).uniform(size=(1, dim_output))
+            v = new_rng(0).uniform(size=(dim_input, 1))
```

Comment:
Maybe two different seeds should be used, such that `w` and `v` are not accidentally the same (or a single `rng` whose `uniform` method is called twice).

Also, it might be better to use the standard normal distribution, as the uniform distribution over (0, 1) doesn't sample negative numbers. Also also, maybe `w` and `v` should be complex when `samples` is complex.